bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitChangeListener.java

```diff
@@ -438,32 +440,51 @@ private HomekitAccessoryType getPrimaryAccessoryType(Item item,
+            return;
+        }
+
         final @Nullable Map<String, Object> itemConfiguration = HomekitAccessoryFactory.getItemConfiguration(item,
                 metadataRegistry);
-        if (accessoryTypes.isEmpty() || !(groups.isEmpty() || groups.stream().noneMatch(g -> g.getBaseItem() == null))
-                || !itemIsForThisBridge(item, itemConfiguration)) {
+        if (!itemIsForThisBridge(item, itemConfiguration))
             return;
-        }
 
         final HomekitAccessoryType primaryAccessoryType = getPrimaryAccessoryType(item, accessoryTypes,
                 itemConfiguration);
         logger.trace("Item {} is a HomeKit accessory of types {}. Primary type is {}", item.getName(), accessoryTypes,
                 primaryAccessoryType);
         final HomekitOHItemProxy itemProxy = new HomekitOHItemProxy(item);
-        final HomekitTaggedItem taggedItem = new HomekitTaggedItem(new HomekitOHItemProxy(item), primaryAccessoryType,
-                itemConfiguration);
+        final HomekitTaggedItem taggedItem = new HomekitTaggedItem(itemProxy, primaryAccessoryType, itemConfiguration);
         try {
             final AbstractHomekitAccessoryImpl accessory = HomekitAccessoryFactory.create(taggedItem, metadataRegistry,
                     updater, settings);
+            if (accessory.isLinkedServiceOnly()) {
+                logger.warn("Item '{}' is a '{}' which must be nested another another accessory.", taggedItem.getName(),
+                        primaryAccessoryType);
+                return;
+            }
 
             accessoryTypes.stream().filter(aType -> !primaryAccessoryType.equals(aType.getKey()))
                     .forEach(additionalAccessoryType -> {
                         final HomekitTaggedItem additionalTaggedItem = new HomekitTaggedItem(itemProxy,
                                 additionalAccessoryType.getKey(), itemConfiguration);
                         try {
-                            final HomekitAccessory additionalAccessory = HomekitAccessoryFactory
+                            final AbstractHomekitAccessoryImpl additionalAccessory = HomekitAccessoryFactory
                                     .create(additionalTaggedItem, metadataRegistry, updater, settings);
+                            // Secondary accessories that don't explicitly specify a name will implicitly
+                            // get a name characteristic based
+                            if (!additionalAccessory.getCharacteristic(HomekitCharacteristicType.NAME).isPresent()) {
+                                try {
+                                    additionalAccessory.addCharacteristic(
+                                            new NameCharacteristic(() -> additionalAccessory.getName()));
+                                } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+                                }
```

Comment:
Can you add a comment here?
```suggestion
                                } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
                                    // Ignore
                                }
```